src/borg/testsuite/repository.py

```diff
@@ -716,339 +767,373 @@ def _subtly_corrupted_hints_setup(self):
+    except PathNotAllowed as e:
+        assert len(e.args) == 1
+        assert e.args[0] == "foo"
+
+    try:
+        remote_repository.call("inject_exception", {"kind": "ObjectNotFound"})
+    except Repository.ObjectNotFound as e:
+        assert len(e.args) == 2
+        assert e.args[0] == s1
+        assert e.args[1] == remote_repository.location.processed
+
+    try:
+        remote_repository.call("inject_exception", {"kind": "InvalidRPCMethod"})
+    except InvalidRPCMethod as e:
+        assert len(e.args) == 1
+        assert e.args[0] == s1
+
+    try:
+        remote_repository.call("inject_exception", {"kind": "divide"})
+    except RemoteRepository.RPCError as e:
+        assert e.unpacked
+        assert e.get_message() == "ZeroDivisionError: integer division or modulo by zero\n"
+        assert e.exception_class == "ZeroDivisionError"
+        assert len(e.exception_full) > 0
+
+
+def test_ssh_cmd(remote_repository):
+    args = _get_mock_args()
+    remote_repository._args = args
+    assert remote_repository.ssh_cmd(Location("ssh://example.com/foo")) == ["ssh", "example.com"]
+    assert remote_repository.ssh_cmd(Location("ssh://user@example.com/foo")) == ["ssh", "user@example.com"]
+    assert remote_repository.ssh_cmd(Location("ssh://user@example.com:1234/foo")) == [
+        "ssh",
+        "-p",
+        "1234",
+        "user@example.com",
+    ]
+    os.environ["BORG_RSH"] = "ssh --foo"
+    assert remote_repository.ssh_cmd(Location("ssh://example.com/foo")) == ["ssh", "--foo", "example.com"]
+
+
+def test_borg_cmd(remote_repository):
+    assert remote_repository.borg_cmd(None, testing=True) == [sys.executable, "-m", "borg", "serve"]
+    args = _get_mock_args()
+    # XXX without next line we get spurious test fails when using pytest-xdist, root cause unknown:
+    logging.getLogger().setLevel(logging.INFO)
+    # note: test logger is on info log level, so --info gets added automagically
+    assert remote_repository.borg_cmd(args, testing=False) == ["borg", "serve", "--info"]
+    args.remote_path = "borg-0.28.2"
+    assert remote_repository.borg_cmd(args, testing=False) == ["borg-0.28.2", "serve", "--info"]
+    args.debug_topics = ["something_client_side", "repository_compaction"]
+    assert remote_repository.borg_cmd(args, testing=False) == [
+        "borg-0.28.2",
+        "serve",
+        "--info",
+        "--debug-topic=borg.debug.repository_compaction",
+    ]
+    args = _get_mock_args()
+    args.storage_quota = 0
+    assert remote_repository.borg_cmd(args, testing=False) == ["borg", "serve", "--info"]
+    args.storage_quota = 314159265
+    assert remote_repository.borg_cmd(args, testing=False) == ["borg", "serve", "--info", "--storage-quota=314159265"]
+    args.rsh = "ssh -i foo"
+    remote_repository._args = args
+    assert remote_repository.ssh_cmd(Location("ssh://example.com/foo")) == ["ssh", "-i", "foo", "example.com"]
+
+
+""" REMOTE REPOSITORY CHECK TESTS"""
```

Comment:
guess one rather wants to use a comment for this.

one docstring on module level (usually at top of the module code) is rather for documenting what the module does.